File: docs/guides/ruby/searchWithRuleContexts.rb
require "algolia"

def get_platform_tag
  # Implement your logic here
  ""
end

begin
  client = Algolia::SearchClient.create("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")

  # get the buyer account information
  platform_tag = get_platform_tag
  search_params = Algolia::Search::SearchParamsObject.new(
    query: "<YOUR_SEARCH_QUERY>",
    rule_contexts: [platform_tag]
  )

  client.search_single_index("<YOUR_INDEX_NAME>", search_params)
rescue Exception => e
  puts("An error occurred: #{e}")
end

File: docs/guides/php/src/searchWithRuleContexts.php
<?php

require __DIR__.'/../vendor/autoload.php';
use Algolia\AlgoliaSearch\Api\SearchClient;
use Algolia\AlgoliaSearch\Model\Search\SearchParamsObject;

$getPlatformTag = function (): string {
    // Implement your logic here
    return '';
};

$client = SearchClient::create('ALGOLIA_APPLICATION_ID', 'ALGOLIA_API_KEY');

// get the buyer account information
$platformTag = $getPlatformTag();
$searchParams = (new SearchParamsObject())
    ->setQuery('<YOUR_SEARCH_QUERY>')
    ->setRuleContexts([$platformTag])
;

$client->searchSingleIndex(
    '<YOUR_INDEX_NAME>',
    $searchParams,
);
